tests/components/FilterSearch.test.tsx

```diff
@@ -0,0 +1,299 @@
+import * as useSynchronizedRequestFunctions from '../../src/hooks/useSynchronizedRequest';
+import userEvent from '@testing-library/user-event';
+import * as searchOperations from '../../src/utils/search-operations';
+import { mockAnswersActions } from '../__utils__/mocks';
+
+jest.mock('@yext/answers-headless-react');
+
+const mockedFilterSearchResponse: FilterSearchResponse = {
+  sections: [{
+    label: 'People',
+    results: [
+      { value: 'John Doe',
+        filter: { fieldId: null, matcher: null, value: 'test' } },
+      { value: 'Jane Doe',
+        filter: { fieldId: null, matcher: null, value: 'test' } }
+    ]
+  },
+  {
+    label: 'Condiments',
+    results: [
+      { value: 'Jam',
+        filter: { fieldId: null, matcher: null, value: 'test' } }
+    ]
+  }],
+  uuid: null
+};
+
+const mockedFilterSearchResponseNoLabels: FilterSearchResponse = {
+  sections: [{
+    results: [
+      { value: 'John Doe',
+        filter: { fieldId: null, matcher: null, value: 'test' } },
+      { value: 'Jane Doe',
+        filter: { fieldId: null, matcher: null, value: 'test' } }
+    ]
+  }],
+  uuid: null
+};
+
+const mockedFilterSearchResponseNoResults: FilterSearchResponse = {
+  sections: [{
+    results:[]
+  }],
+  uuid: null
+};
+
+const mockedExecuteFilterSearch = jest.fn();
+const setFilterOption = jest.fn();
+const setOffset = jest.fn();
+const searchFieldsProp = {
+  fieldApiName: 'name',
+  entityType: 'ce-person'
+};
+
+describe('tests using data with section labels', () => {
+  beforeEach(() => {
+    mockAnswersActions({
+      setFilterOption,
+      setOffset
+    });
+
+    jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+    jest.spyOn(useSynchronizedRequestFunctions, 'useSynchronizedRequest').mockImplementation(() => {
+      return [mockedFilterSearchResponse, mockedExecuteFilterSearch, null];
+    });
+  });
+
+  it('renders the filter search bar and the "Filter" label', () => {
+    render(<FilterSearch searchFields={[searchFieldsProp]}/>);
+
+    const label = 'Filter';
+    const labelElement = screen.getByText(label);
+    const searchBarElement = screen.getAllByRole('textbox');
+
+    expect(labelElement).toBeDefined();
+    expect(searchBarElement.length).toBe(1);
+  });
+
+  it('calls executeFilterSearch every time a charachter is entered', () => {
+    render(<FilterSearch searchFields={[searchFieldsProp]}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    expect(searchBarElement).toBeDefined();
+
+    userEvent.type(searchBarElement, 'J');
+    expect(searchBarElement).toHaveValue('J');
+    expect(mockedExecuteFilterSearch).toBeCalledWith('J');
+
+    userEvent.type(searchBarElement, 'o');
+    expect(searchBarElement).toHaveValue('Jo');
+    expect(mockedExecuteFilterSearch).toBeCalledWith('Jo');
+
+    expect(mockedExecuteFilterSearch.mock.calls.length).toBe(2);
+  });
+
+  it('calls executeFilterSearch when characters changed by backspace', () => {
```

Comment:
I think we can change this test to be just "backspace on an empty text box should not trigger a search" and remove the part where it types "J" and then backspaces J. The backspace part can be moved to 'calls executeFilterSearch every time a charachter is entered'

Reply:
Ok!  Also, for the tests where a search is called whenever a character is entered, I also changed "toHaveBeenCalledWith" to "toHaveBeenLastCalledWith" to be more specific, since we are checking if executeFilterSearch is called with "J", "Jo", then "J" again.